src/main/modules/session/index.ts

```diff
@@ -37,16 +38,20 @@ export class SessionStorage {
   private getStore(app: Application) {
     const redisHost = config.get('session.redis.host');
     if (redisHost) {
-      const client = redis.createClient({
-        host: redisHost as string,
+      const redisClient = createClient({
+        socket: {
+          host: redisHost as string,
+          port: 6380,
+          tls: true,
+          connectTimeout: 15000,
+        },
         password: config.get('session.redis.key') as string,
-        port: 6380,
-        tls: true,
-        connect_timeout: 15000,
       });
 
-      app.locals.redisClient = client;
-      return new RedisStore({ client });
+      redisClient.connect().then(() => {
```

Comment:
if possible, can we do this with async await

Reply:
I tried to do that yesterday but I don't think it's possible as it throws an error
`Top-level 'await' expressions are only allowed when the 'module' option is set to 'esnext'`

Reply:
or a similar error to that I can't remember exactly